consul/resource_consul_acl_role.go

```diff
@@ -298,5 +320,38 @@ func getRole(d *schema.ResourceData, meta interface{}) *consulapi.ACLRole {
 		role.TemplatedPolicies = append(role.TemplatedPolicies, templatedPolicy)
 	}
 
-	return role
+	return role, nil
+}
+
+// getPolicyByIdOrName looks for a policy in Consul first by ID, then by name if
+// it found nothing. It also returns a boolean indicating whether the identifier
+// given is the ID or the name
+func getPolicyByIdOrName(identifier string, client *consulapi.Client, qOpts *consulapi.QueryOptions) (*consulapi.ACLPolicy, bool, error) {
+	policy, _, initialErr := client.ACL().PolicyRead(identifier, qOpts)
+	if policy != nil && initialErr == nil {
```

Comment:
you can rely on both being nil here to detect a not-found use case. I think it would simplify the logic.